wagtail/models/reference_index.py

```diff
@@ -570,6 +634,28 @@ def get_grouped_references_to(cls, object):
         """
         return cls.get_references_to(object).group_by_source_object()
 
+    @classmethod
+    def get_grouped_references_to_in_bulk(cls, objects):
+        """
+        Returns all inbound references for the given objects, grouped by the object
+        they are found on for each given object.
+
+        Args:
+            object: An iterable of model instances to fetch ReferenceIndex records for
+
+        Returns:
+            An dict that maps a model instance to a ReferenceGroups object
+        """
+        references = cls.get_references_to_in_bulk(objects)
+        grouped_references = references.group_by_source_object_in_bulk()
+        return {
+            object: grouped_references.get(
+                (cls._get_base_content_type(object).pk, str(object.pk)),
+                ReferenceGroups([]),
+            )
+            for object in objects
+        }
```

Comment:
I'm not entirely sure how safe it is to use a model instance as a dictionary key. For this use case I think it's pretty safe, given that models use the id for hashing and implement `__eq__` that ensures the equality of the model class.

For extra safety we could use something like `(base_ct_id, object_id)` as the key, but it isn't very practical when you have a list of objects and want to use `get_grouped_references_to_in_bulk` to then get the corresponding `ReferenceGroups` for each object.

If we didn't want to support "instances of mixed models", we could just use a `pk` as the key, but I think it can be quite useful to allow a list of any models (e.g. to be used in a `ReportView`), considering that the `ReferenceIndex` is generic.